### PR TITLE
XML documents were stored incorrectly by WebDAV 

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/4213

### CODE REVIEW

**Comment by joewiz** [comment]
@adamretter I've tested the PR and am seeing no regressions in my use of WebDAV to upload/download/list resources in eXist with these clients on macOS: oXygen, Transmit, and Finder.

File: extensions/webdav/src/test/resources/standalone-webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        <!--
+            Some WebDAV clients send a "Expect: 100-continue" header before
+            uploading body data. Servlet containers (like tomcat and jetty) handle
+            the header in a wrong way, making a client not work OK.
+            Set value to TRUE to restore old behavior (FALSE is the new default
+            value, hardcoded in MiltonWebDAVServlet).
+        -->
+        <!--
+            <init-param>
+            <param-name>enable.expect.continue</param-name>
```
Comment (line-o) [inline extensions/webdav/src/test/resources/standalone-webapp/WEB-INF/web.xml:84]: Are there any advantages in allowing this again? 
And why did it work prior to version 5.4.0?

File: extensions/webdav/src/test/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        <RollingRandomAccessFile name="exist.xmldb" filePattern="${logs}/xmldb.${rollover.file.pattern}.log.gz" fileName="${logs}/xmldb.log">
+            <Policies>
+                <SizeBasedTriggeringPolicy size="${rollover.max.size}"/>
+            </Policies>
+            <DefaultRolloverStrategy max="${rollover.max}"/>
+            <PatternLayout pattern="${exist.file.pattern}"/>
+        </RollingRandomAccessFile>
+
+        <RollingRandomAccessFile name="exist.xmlrpc" filePattern="${logs}/xmlrpc.${rollover.file.pattern}.log.gz" fileName="${logs}/xmlrpc.log">
```
Comment (line-o) [inline extensions/webdav/src/test/resources/log4j2.xml:52]: Do we need the XML-RPC logs for webdav testing? Can this and others be safely removed? Would that speed up testing (even a little bit)?

**Comment by line-o** [comment]
[Newly added tests are currently failing](https://github.com/eXist-db/exist/runs/5118298188?check_suite_focus=true#step:5:9793) » FileSystemNotFound

**Comment by line-o** [comment]
Codacy complains about the tests not containing an explicit assertion (`assert` or `fail`).
I believe this is a valid complaint and would like to ask to refactor the tests to make it obvious where the assertion is taking place within each case.

File: extensions/webdav/src/test/resources/standalone-webapp/WEB-INF/web.xml (hunk 1 above)
Comment (adamretter) [reply extensions/webdav/src/test/resources/standalone-webapp/WEB-INF/web.xml:84]: This is not related to the issue or the fix. i.e. this is unchanged - this is just config needed to run eXist-db WebServer in the WebDAV tests. This is used in other tests too.

File: extensions/webdav/src/test/resources/log4j2.xml (hunk 2 above)
Comment (adamretter) [reply extensions/webdav/src/test/resources/log4j2.xml:52]: This is unlikely to make any difference. It is simpler to have consistent log4j2.xml files, than lots of different ones.

**Comment by adamretter** [comment]
> [Newly added tests are currently failing](https://github.com/eXist-db/exist/runs/5118298188?check_suite_focus=true#step:5:9793) » FileSystemNotFound

@line-o That is now fixed.

**Comment by adamretter** [comment]
> Codacy complains about the tests not containing an explicit assertion (assert or fail).
> I believe this is a valid complaint

I am afraid that Codacy is wrong here. If you check how the tests work, the assertions are common to all test methods and so are in the common function (which is perfectly valid) - Codacy is failing to trace the code correctly.

**Review by dizzzz — APPROVED** [review]

**Comment by line-o** [comment]
Renaming the function to include the keyword assert might be enough for codacy - and humans - to pick it up.

**Pull request merged**